src/transactionsFactories/transferTransactionsFactory.ts

```diff
@@ -17,23 +36,66 @@ interface IConfig {
     gasLimitMultiESDTNFTTransfer: bigint;
 }
 
-interface TokenComputer {
+interface ITokenComputer {
     isFungible(token: Token): boolean;
 }
 
+interface IGasEstimator {
+    forEGLDTransfer(dataLength: number): number;
+    forESDTTransfer(dataLength: number): number;
+    forESDTNFTTransfer(dataLength: number): number;
+    forMultiESDTNFTTransfer(dataLength: number, numTransfers: number): number;
+}
+
 /**
  * Use this class to create transactions for native token transfers (EGLD) or custom tokens transfers (ESDT/NTF/MetaESDT).
  * This name is only temporary, the class will be renamed to `TransferTransactionsFactory`.
  */
-export class NextTransferTransactionsFactory {
-    private readonly config: IConfig;
-    private readonly dataArgsBuilder: TokenTransfersDataBuilder;
-    private readonly tokenComputer: TokenComputer;
-
-    constructor(options: { config: IConfig; tokenComputer: TokenComputer }) {
-        this.config = options.config;
-        this.tokenComputer = options.tokenComputer;
-        this.dataArgsBuilder = new TokenTransfersDataBuilder();
+export class TransferTransactionsFactory {
+    private readonly config?: IConfig;
+    private readonly dataArgsBuilder?: TokenTransfersDataBuilder;
+    private readonly tokenComputer?: ITokenComputer;
+    private readonly gasEstimator?: IGasEstimator;
+
+    constructor(options: IGasEstimator | { config: IConfig; tokenComputer: ITokenComputer }) {
+        if (this.isGasEstimator(options)) {
+            this.gasEstimator = options;
+        } else {
+            this.config = options.config;
+            this.tokenComputer = options.tokenComputer;
+            this.dataArgsBuilder = new TokenTransfersDataBuilder();
+        }
+    }
+
+    private isGasEstimator(options: any): options is IGasEstimator {
+        return (
+            typeof options === "object" &&
+            typeof options.forEGLDTransfer === "function" &&
+            typeof options.forESDTTransfer === "function" &&
+            typeof options.forESDTNFTTransfer === "function" &&
+            typeof options.forMultiESDTNFTTransfer === "function"
+        );
+    }
+
+    private isGasEstimatorDefined(): boolean {
+        if (this.gasEstimator === undefined) {
+            return false;
+        }
+        return this.isGasEstimator(this.gasEstimator);
```

Comment:
Maybe `this.gasEstimator === undefined` is a "sufficient" check (covers the flow where the user called the constructor with gas estimator)?

Reply:
That should be enough. Removed.